### PR TITLE
Fix handling of empty SCS17/SCS18 messages

### PR DESCRIPTION
**Pull request by rm5248**
The code for handling empty SCS_17 and SCS_18 messages doesn't work correctly.  Fix it so that it does.

Because this is non-conforming behavior, added a new flag to optionally enable it.

### CODE REVIEW

**Review by sidcha — CHANGES_REQUESTED** [review]
Thanks for the fix. It would help explaining what is broken (how this patch fixes it) in the commit message; especially for fixes like this. I'm not actively working on LibOSDP all the time and rely heavily on `git blame` to rebuild context.

**Review by sidcha — APPROVED** [review]
LGTM, thanks for the PR.

File: README.md [hunk 1]
```diff
@@ -162,8 +162,7 @@ is how you can run them:
 ```sh
 mkdir build && cd build
 cmake ..
-make python_install
-make check
+make check-ut
```
Comment (Sympatron) [inline README.md:165]: Was this change intended?

Comment (sidcha) [reply README.md:165]: This hunk is not related to this PR (commit message) looked odd to be here; but I'm practicing to avoid nit-picking in reviews :)

But the change itself is [sound](https://github.com/goToMain/libosdp/blob/master/tests/unit-tests/CMakeLists.txt#L45).
